projects/i18n/switch/language-switcher-token.ts

```diff
@@ -0,0 +1,20 @@
+import type {Provider} from '@angular/core';
+import {TUI_LANGUAGE} from '@taiga-ui/i18n/tools';
+
+import {TUI_LANGUAGE_CHUNK_LOADER} from './language-chunk-loader-token';
+import {TuiLanguageSwitcher} from './language-switcher';
+import {TuiLangChunkLoader} from './options';
+
+export function tuiLanguageSwitcher(loader: TuiLangChunkLoader): Provider[] {
+    return [
+        {
+            provide: TUI_LANGUAGE_CHUNK_LOADER,
+            useFactory: () => loader,
+        },
+        {
+            provide: TUI_LANGUAGE,
+            deps: [TuiLanguageSwitcher],
+            useFactory: (switcher: TuiLanguageSwitcher) => switcher.asObservable(),
```

Comment:
I don't think it's necessary to cast it. We can write `useExisting`

Reply:
fixed